compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -387,13 +398,31 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
 
     @Override
     public void visit(BLangTupleTypeNode tupleTypeNode) {
+        BTypeSymbol typeSymbol = tupleTypeNode.getBType().tsymbol;
+        BSymbol owner = typeSymbol.owner;
+        desugarFieldAnnotations(owner, typeSymbol, tupleTypeNode.members, tupleTypeNode.pos);
         List<BLangSimpleVariable> rewrittenMembers = new ArrayList<>();
         tupleTypeNode.members.forEach(member -> rewrittenMembers.add(rewrite(member, env)));
         tupleTypeNode.members = rewrittenMembers;
         tupleTypeNode.restParamType = rewrite(tupleTypeNode.restParamType, env);
         result = tupleTypeNode;
     }
 
+    private void desugarFieldAnnotations(BSymbol owner, BTypeSymbol typeSymbol, List<BLangSimpleVariable> fields,
+                                         Location pos) {
+        if (owner.getKind() != SymbolKind.PACKAGE) {
+            owner = getOwner(env);
+            BLangLambdaFunction lambdaFunction = annotationDesugar.defineFieldAnnotations(fields, pos, env.enclPkg, env,
+                                                                                          typeSymbol.pkgID, owner);
+            if (lambdaFunction != null) {
+                boolean isPackageLevel = owner.getKind() == SymbolKind.PACKAGE;
+                BInvokableSymbol invokableSymbol = createSimpleVariable(lambdaFunction.function, lambdaFunction,
+                                                                        isPackageLevel);
+                typeSymbol.annotations = createSimpleVariable(invokableSymbol, isPackageLevel);
```

Comment:
Is the name of the last parameters of these functions ideal? Seems to be `isAnnotationClosure`?

Reply:
Change to `isPackageLevelAnnotationClosure`